package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,200 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding: utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the Lesser GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Base guesser classes --- :mod:`MDAnalysis.guesser.base`
+================================================================
+
+Derive context-specific guesser classes from the base class in this module.
+
+Classes
+-------
+
+.. autoclass:: GuesserBase
+   :members:
+   :inherited-members:
+
+.. autofunction:: get_guesser
+
+"""
+from .. import _GUESSERS
+import numpy as np
+from .. import _TOPOLOGY_ATTRS
+import logging
+from typing import Dict
+import copy
+
+logger = logging.getLogger("MDAnalysis.guesser.base")
+
+
+class _GuesserMeta(type):
+    """Internal: guesser classes registration
+
+    When classes which inherit from GuesserBase are *defined*
+    this metaclass makes it known to MDAnalysis.  'context'
+    attribute  are read:
+     - `context` defines the context of the guesser class for example:
+       forcefield specific context  as MartiniGuesser
+       and file specific context as PDBGuesser.
+
+    Eg::
+
+      class FooGuesser(GuesserBase):
+          format = 'foo'
+
+    .. versionadded:: 2.8.0
+    """
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+
+        _GUESSERS[classdict['context'].upper()] = cls
+
+
+class GuesserBase(metaclass=_GuesserMeta):
+    """Base class for context-specific guessers to inherit from
+
+    Parameters
+    ----------
+    universe : Universe, optional
+        Supply a Universe to the Guesser. This then becomes the source of atom
+        attributes to be used in guessing processes. (this is relevant to how
+        the universe's guess_TopologyAttrs API works.
+        See :meth:`~MDAnalysis.core.universe.Universe.guess_TopologyAttrs`).
+    **kwargs: to pass additional data to the guesser that can be used with
```

Comment:
```suggestion
    **kwargs : dict, optional
        To pass additional data to the guesser that can be used with
```
Currently this shows up in the docs as "**kwargs (to pass additional data to the guesser that can be used with) – different methods." rather than "**kwargs - to pass additional data to the guesser that can be used with different methods."